addons/mail/static/src/components/message_action_list/message_action_list.js

```diff
@@ -0,0 +1,40 @@
+/** @odoo-module */
+import { registerMessagingComponent } from '@mail/utils/messaging_component';
+import { markEventHandled } from '@mail/utils/utils';
+
+const { Component } = owl;
+export class MessageActionList extends Component {
+
+    /**
+     * @returns {mail.message}
+     */
+    get message() {
+        return this.messaging && this.messaging.models['mail.message'].get(this.props.messageLocalId);
+    }
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Discards the composer when clicking away.
+     *
+     * @private
+     * @param {MouseEvent} ev
+     */
+    _onClickReplyTo(ev) {
+        markEventHandled(ev, 'MessageActionList.replyTo');
+        this.message.replyTo();
+    }
```

Comment:
move to model

Reply:
Are you saying that the model should be handling DOM events? The actual replyTo implementation is already in models.

Reply:
> Are you saying that the model should be handling DOM events? The actual replyTo implementation is already in models.

We want to dispatch user interaction to model ASAP and keep most logic in model. This means **the click on reply** should also be in model too.

It's alright to manage DOM event in models. There are bad stuffs in the payload (e.g. DOM references), but good practice we follow consists of not making use of them, so you can see these DOM events like records that are modelling events from user interaction on the UI. These are not records made from our models and fields, but it's reasonable to rely on them for the time being. If these were actual records from our models, we could make discuss code much more declarative, so less code and more robust behaviour overall (e.g. no need to mark event as handled by hand, no need to write function to imperatively dispatch event, no more function handler but just computed fields, etc.).